trieste/utils/mo_objectives.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains toy multi-objective functions, useful for experimentation.
+"""
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from functools import partial
+
+import tensorflow as tf
+
+from ..type import TensorType
+
+
+class MultiObjectiveTestProblem(ABC):
+    """base class for synthetic multi-objective test functions"""
+
+    @abstractmethod
+    def prepare_benchmark(self):
```

Comment:
This needs a return type, and a docstring! Is this returning an objective function of the form `Callable[[TensorType], TensorType]`? 

Also, is the word "prepare" necessary here? And is this a "benchmark" or just the "objective"?

Reply:
Maybe not necessary, used `objective` instead.